File: app/src/main/java/dot/ga/gov/customdialog/HomeActivity.java
package dot.ga.gov.customdialog;


import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.*;

public class HomeActivity extends AppCompatActivity implements DataAdapter.RecyclerViewItemClickListener {
    Button clickButton;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);
        clickButton = (Button) findViewById(R.id.button);
    }

    CustomListViewDialog customDialog;

    public void clickHere(View view) {
        final String[] items = {"Apple Apple Apple ", "Banana", "Orange", "Grapes", "Apple", "Banana", "Orange", "Grapes", "Apple", "Banana", "Orange", "Grapes", "Apple", "Banana", "Orange", "Grapes", "Apple", "Banana", "Orange", "Grapes"};
//        final String[] items = { "Apple", "Banana", "Orange", "Grapes", "Apple", "Banana", "Orange", "Grapes"};

        DataAdapter dataAdapter = new DataAdapter(items, this);
        customDialog = new CustomListViewDialog(HomeActivity.this, dataAdapter);

        customDialog.show();
        customDialog.setCanceledOnTouchOutside(false);


    }

    @Override
    public void clickOnItem(String data) {
        clickButton.setText(data);

        if (customDialog != null){
            customDialog.dismiss();
        }
    }
}

File: app/src/main/java/dot/ga/gov/customdialog/HomeActivity.kt
package dot.ga.gov.customdialog


import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View

import kotlinx.android.synthetic.main.activity_home.*

class HomeActivity : AppCompatActivity(), DataAdapter.RecyclerViewItemClickListener {


    internal var customDialog: CustomListViewDialog? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

    }

    fun clickHere(view: View) {
        val items = arrayOf(
            "Apple Apple Apple ",
            "Banana",
            "Orange",
            "Grapes",
            "Apple",
            "Banana",
            "Orange",
            "Grapes",
            "Apple",
            "Banana",
            "Orange",
            "Grapes",
            "Apple",
            "Banana",
            "Orange",
            "Grapes",
            "Apple",
            "Banana",
            "Orange",
            "Grapes"
        )


        val dataAdapter = DataAdapter(items, this)
        customDialog = CustomListViewDialog(this@HomeActivity, dataAdapter)

        //if we know that the particular variable not null any time ,we can assign !! (not null operator ), then  it won't check for null, if it becomes null, it willthrow exception
        customDialog!!.show()
        customDialog!!.setCanceledOnTouchOutside(false)


    }

    override fun clickOnItem(data: String) {
        //Synthetic property without calling findViewById() method and supports view caching to improve performance.
        button.text = data

        if (customDialog != null) {
            customDialog!!.dismiss()
        }
    }
}
